test/users/npadmana/blas/BLAS.chpl

```diff
@@ -0,0 +1,729 @@
+          alpha1, A, _ldA, B, _ldB, beta1, C,_ldC);
+      }
+      otherwise {
+        halt("Unknown type in her2k");
+      }
+    }
+
+  }
+
+  /* Wrapper for the TRMM routines
+  */
+  proc trmm(A : [?Adom],  B : [?Bdom],
+    alpha,
+    uplo : Uplo = Uplo.Upper,  trans : Op = Op.N,
+    side : Side = Side.Left, diag : Diag = Diag.NonUnit,
+    order : Order = Order.Row,
+    ldA : int = 0,  ldB : int = 0)
+    where (Adom.rank == 2) && (Bdom.rank==2)
+  {
+    // Types
+    type eltType = A.eltType;
+
+    // Determine sizes
+    var m = Bdom.dim(1).size : c_int,
+        n = Bdom.dim(2).size : c_int;
+
+    // Set strides if necessary
+    var _ldA = getLeadingDim(Adom, order, ldA),
+        _ldB = getLeadingDim(Bdom, order, ldB);
+
+    select eltType {
+      when real(32) {
+        C_BLAS.cblas_strmm(order, side, uplo, trans, diag,
+          m, n, alpha, A, _ldA, B, _ldB);
+      }
+      when real(64) {
+        C_BLAS.cblas_dtrmm(order, side, uplo, trans, diag,
+          m, n, alpha, A, _ldA, B, _ldB);
+      }
+      when complex(64) {
+        var alpha1 = alpha : complex(64);
+        C_BLAS.cblas_ctrmm(order, side, uplo, trans, diag,
+          m, n, alpha1, A, _ldA, B, _ldB);
+      }
+      when complex(128) {
+        var alpha1 = alpha : complex(128);
+        C_BLAS.cblas_ztrmm(order, side, uplo, trans, diag,
+          m, n, alpha1, A, _ldA, B, _ldB);
+      }
+      otherwise {
+        halt("Unknown type in trmm");
+      }
+    }
+
+  }
+
+  /* Wrapper for the TRSM routines
+  */
+  proc trsm(A : [?Adom],  B : [?Bdom],
+    alpha,
+    uplo : Uplo = Uplo.Upper,  trans : Op = Op.N,
+    side : Side = Side.Left, diag : Diag = Diag.NonUnit,
+    order : Order = Order.Row,
+    ldA : int = 0,  ldB : int = 0)
+    where (Adom.rank == 2) && (Bdom.rank==2)
+  {
+    // Types
+    type eltType = A.eltType;
+
+    // Determine sizes
+    var m = Bdom.dim(1).size : c_int,
+        n = Bdom.dim(2).size : c_int;
+
+    // Set strides if necessary
+    var _ldA = getLeadingDim(Adom, order, ldA),
+        _ldB = getLeadingDim(Bdom, order, ldB);
+
+    select eltType {
+      when real(32) {
+        C_BLAS.cblas_strsm(order, side, uplo, trans, diag,
+          m, n, alpha, A, _ldA, B, _ldB);
+      }
+      when real(64) {
+        C_BLAS.cblas_dtrsm(order, side, uplo, trans, diag,
+          m, n, alpha, A, _ldA, B, _ldB);
+      }
+      when complex(64) {
+        var alpha1 = alpha : complex(64);
+        C_BLAS.cblas_ctrsm(order, side, uplo, trans, diag,
+          m, n, alpha1, A, _ldA, B, _ldB);
+      }
+      when complex(128) {
+        var alpha1 = alpha : complex(128);
+        C_BLAS.cblas_ztrsm(order, side, uplo, trans, diag,
+          m, n, alpha1, A, _ldA, B, _ldB);
+      }
+      otherwise {
+        halt("Unknown type in trsm");
+      }
+    }
+
+  }
+
+
+  // Helper function
+  pragma "no doc"
+  inline proc getLeadingDim(Adom : domain(2), order : Order, ldA : int) : c_int {
```

Comment:
Dumb question -- what's the best way to make a function private?

On Wednesday, July 6, 2016, Ben Harshbarger <notifications@github.com>
wrote:

> In test/users/npadmana/blas/BLAS.chpl
> <https://github.com/chapel-lang/chapel/pull/3931#discussion_r69824567>:
>
> > +      when complex(128) {
> > +        var alpha1 = alpha : complex(128);
> > +        C_BLAS.cblas_ztrsm(order, side, uplo, trans, diag,
> > +          m, n, alpha1, A, _ldA, B, _ldB);
> > +      }
> > +      otherwise {
> > +        halt("Unknown type in trsm");
> > +      }
> > +    }
> > +
> > +  }
> > +
> > +
> > +  // Helper function
> > +  pragma "no doc"
> > +  inline proc getLeadingDim(Adom : domain(2), order : Order, ldA : int) : c_int {
>
> Could we make this a private function, just to keep the namespace clean?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/chapel-lang/chapel/pull/3931/files/31ac9b15b5516bdd5f678bbee109f9b2b4153d6d#r69824567>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAkfaiqbQdpUEEB3c7nMFfsprSXo2NJqks5qTC68gaJpZM4IpNSS>
> .
>


-- 
---------------------------------
Nikhil Padmanabhan
nikhil.padmanabhan@yale.edu